### PR TITLE
chore(ssa refactor): Adds basic program with empty body to experimental-ssa test corpus 

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

This adds the program from #1403 into the test corpus.

## Problem\*

Resolves 

## Summary\*

This PR sets out to

### Example

Before:

```

```

After:

```

```

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
Copying the CI output:


```
 ---- tests::noir_integration_ssa_refactor stdout ----
 Running test simple_program_no_body
 Initial SSA:
 fn main f0 {
   b0(v0: Field, v1: Field):
     return unit 0
 }
 After Inlining:
 fn main f1 {
   b0(v0: Field, v1: Field):
     return unit 0
 }
 After Unrolling:
 fn main f1 {
   b0(v0: Field, v1: Field):
     return unit 0
 }
 After Simplifying:
 fn main f1 {
   b0(v0: Field, v1: Field):
     return unit 0
 }
 After Flattening:
 fn main f1 {
   b0(v0: Field, v1: Field):
     return unit 0
 }
 After Mem2Reg:
 fn main f1 {
   b0(v0: Field, v1: Field):
     return unit 0
 }
 
 thread 'main' panicked at 'ICE: Return of value not yet encountered', crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs:113:28
```

**Comment by kevaundray** [comment]
Error is occuring because when we try to do acir_gen for the return instruction, we:

- Get the return values
- Check to see if we've acir_gen'd this return value and panic if not

The second point assumes that the return value will always be a variable that was encountered before, whereas  `return unit 0` is a constant that the acir_gen pass has never seen before

**Comment by jfecher** [comment]
Looks good to me but we should avoid merging until the issue in acir-gen is fixed

**Comment by kevaundray** [comment]
> ssa-gen.

Yep will debug this -- I think it would also trigger, if this returns any ValueId which was not used in another part of the program, so I imagine just returning a constant would also trigger this bug

**Comment by kevaundray** [comment]
Seems there is another error in acir_gen:

```
CompiledProgram {
    circuit: current witness index : 3
    public parameters indices : [1, 2]
    return value indices : [3]
    EXPR [ (-1, _3) 0 ]
    ,
    abi: Abi {
        parameters: [
            AbiParameter {
                name: "_x",
                typ: Field,
                visibility: Private,
            },
            AbiParameter {
                name: "_y",
                typ: Field,
                visibility: Public,
            },
        ],
        param_witnesses: {
            "_x": [
                Witness(
                    1,
                ),
            ],
            "_y": [
                Witness(
                    2,
                ),
            ],
        },
        return_type: None,
        return_witnesses: [
            Witness(
                3,
            ),
        ],
    },
}

Serialized transcript does not contain the required number of bytes
```

> Serialized transcript does not contain the required number of bytes

This error really only happens in the barretenberg when there is a discrepancy between the circuit or proving system. Since the proving system has not changed, then we can then deduce that something is malformed about the circuit.

In particular, that error arises when there is a mismatch between public inputs.
----

To fix:

- If the return is `unit 0` then we should treat this as not having a return parameter. We should then see `return value indices : []` ie no returns. This should also get rid of ` EXPR [ (-1, _3) 0 ]`.

- The next issue is `public parameters indices : [1, 2]` we should have ``public parameters indices : [2]` indicating that there is one public parameter and it is the second parameter which should have index 2

**Comment by kevaundray** [comment]
Also going to block this PR until we check for:


```
fn main() -> pub Field {
 return <constant>;
}
```

To make sure we don't mess something up there

**Comment by kevaundray** [comment]
Tested this with a constant return and that also passes :) This passes for me locally so just waiting for it to pass on CI

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs [hunk 1]
```diff
@@ -104,17 +104,41 @@ impl Context {
...
+                return Type::Unit == *typ;
+            }
+
+            // If its not a numeric constant, then it cannot be the void/unit type
+            false
+        }
+
+        if is_return_unit_type(return_values, dfg) {
+            return;
+        }
```
Comment (jfecher) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs:137]:
```suggestion
        let is_return_unit_type = return_values.len() == 1 && dfg.type_of_value(return_values[0]) == Type::Unit;
        if is_return_unit_type {
            return;
        }
```

**Review by jfecher — APPROVED** [review]

**Pull request merged**